.github/workflows/test.yml

```diff
@@ -49,16 +49,20 @@ jobs:
         run: go mod tidy -go=1.19 -compat=1.17 && git diff --exit-code
         if: ${{ matrix.go == '1.19' }}
       - name: Test
-        run: go test -count=1 ./...
-      - name: Test (race)
-        run: go test -count=1 -race ./... -coverprofile=coverage.out -covermode=atomic
+        run: make test
+      - name: Test (with race detection)
+        run: make test-race
         # The race detector adds considerable runtime overhead. To save time on
         # pull requests, only run this step for a single job in the matrix. For
         # all other workflow triggers (e.g., pushes to a release branch) run
         # this step for the whole matrix.
         if: ${{ github.event_name != 'pull_request' || (matrix.go == '1.19' && matrix.os == 'ubuntu') }}
```

Comment:
Could we only omit `-race` for PRs when the runner being windows?
We have to be careful to produce the same amount of coverage reports in any case though.

Reply:
We don't generate any coverage reports in this step.
"Collect coverage" (the next one) runs for all platforms all the time.

Reply:
For my confirmation, looking at the matrix

- `make test` runs tests for all Go versions on ubuntu/windows/macOS
- `make test-coverage` runs tests for all Go versions on ubuntu/windows/macOS and produces coverage reports
- `make test-race` runs tests for all Go versions on ubuntu/windows/macOS, if not in a PR
- `make test-race` (in `test-gopath`) runs tests for all Go versions on ubuntu

Reply:
* `make test` runs tests for all Go versions on ubuntu/windows/macOS
    Yes   

* `make test-coverage` runs tests for all Go versions on ubuntu/windows/macOS and produces coverage reports
    Yes 
* `make test-race runs` tests for all Go versions on ubuntu/windows/macOS, if not in a PR
    If not in a PR -- yes, if PR -- only for 1.19 and Linux.
* make test-race (in test-gopath) runs tests for all Go versions on ubuntu
    Only for the default `actions/setup-go@v3` Go version, which now seems to be 1.18. We can pin that more explicitly.

Reply:
So we could combine `make test` & `make test-coverage` in the CI? Or is there something I'm missing?

Reply:
We could, yes. 
I was thinking about this more like about separate stages (first -- just run tests, second -- run them with race detection, third -- run with coverage), and generally, I see the value in running the tests more than once, at the very least to identify flaky tests (e.g. those that implicitly rely on randomness of map iteration, or, say, order of Baggage elements). But that can be easily changed by passing `-count 10` to `go test` (not doing it here, though)
So if you feel strongly about this, I can merge `test` and `test-coverage` steps.

Reply:
It's ok, was just curious about the reasoning.